### PR TITLE
CSV-driven convolution test pipeline

### PR DESCRIPTION
**Pull request by JH-Leon-KIM-AMD**
## Summary
  Generate convolution test datasets from real PyTorch models by capturing MIOpen commands.

### Pipeline Flow
  PyTorch Models → MIOpen Commands → CSV Test Cases → CK GTest

  ### Main Script
  **generate_test_dataset.sh**: Generate convolution test datasets
  - Check for existing files before running
  - Configurable MAX_ITERATIONS (0=unlimited, 10=quick test, default=20)
  - Combine individual CSVs into final datasets

  ### Components
  1. **generate_model_configs.py**: Generate model configurations
     - 2D models: ResNet, VGG, MobileNet, DenseNet, AlexNet, GoogleNet, 
  SqueezeNet, ShuffleNet
     - 3D models: R3D, MC3, R(2+1)D
     - Input sizes: 64x64 to 512x512, plus rectangular inputs
     - Single precision (fp32) only

  2. **run_model_with_miopen.py**: Execute PyTorch models
     - Run models with MIOPEN_ENABLE_LOGGING_CMD=1
     - Capture convolution commands to log files

  3. **miopen_to_csv.py**: Convert MIOpen commands to CSV
     - Parse MIOpen driver commands
     - Generate CK test format: NDim, Groups, BatchSize, Channels, Kernel, 
  Stride, Padding

  ### Output (test cases for test_grouped_convnd_fwd_dataset_xdl)
  - conv_test_set_2d_dataset.csv: 2D convolution test cases
  - conv_test_set_3d_dataset.csv: 3D convolution test cases
  - Format: CSV without header row (header as comment)

  ### Testing
  Tests run via test_grouped_convnd_fwd_dataset_xdl with multiple precisions 
  (fp32/fp16/bf16/int8).

  ## Usage
  ```bash
  cd composable_kernel/test_data
  ./generate_test_dataset.sh

  cd ../script
  make -j64 test_grouped_convnd_fwd_dataset_xdl
  ./bin/test_grouped_convnd_fwd_dataset_xdl
```

  Files
  - Added: generate_model_configs.py, test_csv_parsing.sh
  - Modified: generate_test_dataset.sh, miopen_to_csv.py,
  run_model_with_miopen.py
  - Removed: generate_systematic_cases.py

## First PR 
- Add test_grouped_convnd_fwd_dataset_xdl.cpp with CSV reader functionality
- Add complete dataset generation toolchain in test_data/
- Add Jenkins integration with RUN_CONV_COMPREHENSIVE_DATASET parameter
- Ready for comprehensive convolution testing with scalable datasets

### CODE REVIEW

File: test_data/run_model_with_miopen.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    with torch.no_grad():
+        for i in range(3):
+            print(f"Iteration {i+1}")
+            output = model(input_tensor)
+            print(f"Output shape: {output.shape}")
+    
+    print("Done! MIOpen commands logged to stderr")
+
+if __name__ == "__main__":
+    main()
```
Comment (bartekxk) [inline test_data/run_model_with_miopen.py]: please add end line

File: test_data/run_model_with_miopen.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        self.block2 = MobileNetBlock(64, 128, stride=2)
+        self.block3 = MobileNetBlock(128, 256, stride=2)
+        
+    def forward(self, x):
+        x = F.relu(self.conv1(x))
+        x = self.block1(x)
+        x = self.block2(x)
+        x = self.block3(x)
+        return x
+
```
Comment (bartekxk) [inline test_data/run_model_with_miopen.py]: Maybe we can use torchvision.models?

File: test_data/miopen_to_csv.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+            write_csv_cases(test_cases_2d, args.output_2d, 2)
+        
+        if test_cases_3d:
+            write_csv_cases(test_cases_3d, args.output_3d, 3)
+    
+    print("Conversion completed!")
+    return 0
+
+if __name__ == "__main__":
+    exit(main())
```
Comment (bartekxk) [inline test_data/miopen_to_csv.py]: end line required

File: test_data/miopen_to_csv.py [hunk 4]
```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+"""
```
Comment (bartekxk) [inline test_data/miopen_to_csv.py:2]: Maybe we can just store miopen driver commands and run them using ../script/convert_miopen_driver_to_profiler.py ?

File: test_data/run_model_with_miopen.py (hunk 1 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/run_model_with_miopen.py]: Thank you for the feedback! I've addressed your comments.

File: test_data/run_model_with_miopen.py (hunk 2 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/run_model_with_miopen.py]: Thank you for the feedback! 
Now pipeline creates configurations from the torchvision models.

File: test_data/miopen_to_csv.py (hunk 3 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/miopen_to_csv.py]: Thank you for the feedback!

File: test_data/miopen_to_csv.py (hunk 4 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/miopen_to_csv.py:2]: I just finished this task before reading this comment. 😅
The pipeline updated but works like previous way.
**PyTorch Models → MIOpen Commands → CSV Test Cases → CK GTest**

File: test_data/generate_model_configs.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+"""
+Generate Model Configuration Combinations for MIOpen Testing
+
```
Comment (bartekxk) [inline test_data/generate_model_configs.py:4]: Random place:
Can we remove:
 conv_test_set_2d_dataset.csv: 2D convolution test cases
conv_test_set_3d_dataset.csv: 3D convolution test cases
and generate them in the fly?

File: test_data/generate_model_configs.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+"""
+Generate Model Configuration Combinations for MIOpen Testing
+
+This script generates all possible combinations of model parameters
+and saves them as CSV files that can be read by the shell script.
+"""
```
Comment (bartekxk) [inline test_data/generate_model_configs.py:7]: Generally in MIOpen C is equal to the number of Channels including all groups so it is C * G. In the Ck C is channel per group.
So if you use C from MIOpen cmd then it should be divided by G.

File: test_data/test.txt [hunk 7]
```diff
@@ -0,0 +1,102 @@
+MODEL_CONFIGS["resnet18_std"]="--model resnet18 --batch-size 16 --input-size 224"
+MODEL_CONFIGS["resnet18_large"]="--model resnet18 --batch-size 32 --height 256 --width 256"
+MODEL_CONFIGS["resnet18_small"]="--model resnet18 --batch-size 8 --height 128 --width 128"
+MODEL_CONFIGS["resnet18_rect"]="--model resnet18 --batch-size 16 --height 224 --width 320"
```
Comment (bartekxk) [inline test_data/test.txt]: Do we need this file? Can we generate it in the fly?

File: test_data/test_csv_parsing.sh [hunk 8]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+OUTPUT_DIR="generated_datasets"
+
```
Comment (bartekxk) [inline test_data/test_csv_parsing.sh]: Do we need this script?

File: test_data/generate_model_configs.py (hunk 5 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/generate_model_configs.py:4]: Yes they will be created in the fly. I will remove it.

File: test_data/generate_model_configs.py (hunk 6 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/generate_model_configs.py:7]: Thank you.
I added this code in miopen_to_csv.py
```
out_channels = out_channels_total // groups  # CK format: channels per group
in_channels = in_channels_total // groups    # CK format: channels per group

```
So, this fix ensures:
  - MIOpen format (total channels): C = 96, K = 128, G = 4
  - CK format (per-group channels): C = 24, K = 32, G = 4

File: test_data/test.txt (hunk 7 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/test.txt]: Sorry this is test output, I will remove it.

File: test_data/test_csv_parsing.sh (hunk 8 above)
Comment (JH-Leon-KIM-AMD) [reply test_data/test_csv_parsing.sh]: Sorry, this is test purpose. I will remove it. Thank you!

**Review by bartekxk — DISMISSED** [review]
LGTM, please update the PR and rerun CI. After pass feel free to merge

**Review by bartekxk — APPROVED** [review]

**Review by bartekxk — APPROVED** [review]
